checks/evaluation/branch_protection.go

```diff
@@ -60,41 +70,117 @@ const (
 )
 
 // BranchProtection runs Branch-Protection check.
-func BranchProtection(name string, dl checker.DetailLogger,
-	r *checker.BranchProtectionsData,
+func BranchProtection(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
 ) checker.CheckResult {
-	var scores []levelScore
+	expectedProbes := []string{
+		blocksDeleteOnBranches.Probe,
+		blocksForcePushOnBranches.Probe,
+		branchProtectionAppliesToAdmins.Probe,
+		dismissesStaleReviews.Probe,
+		requiresApproversForPullRequests.Probe,
+		requiresCodeOwnersReview.Probe,
+		requiresLastPushApproval.Probe,
+		requiresUpToDateBranches.Probe,
+		runsStatusChecksBeforeMerging.Probe,
+		requiresPRsToChangeCode.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	branchScores := make(map[string]*levelScore)
+	warnedBranches := make([]string, 0)
 
-	// Check protections on all the branches.
-	for i := range r.Branches {
-		var score levelScore
-		b := r.Branches[i]
+	for i := range findings {
+		f := &findings[i]
+		if f.Outcome == finding.OutcomeNotApplicable {
+			return checker.CreateInconclusiveResult(name, "unable to detect any development/release branches")
+		}
 
+		var branchName string
+		for k := range f.Values {
+			if k == "branchProtected" || k == "numberOfRequiredReviewers" || k == "codeownersFiles" {
+				continue
+			}
+			branchName = k
+		}
 		// Protected field only indates that the branch matches
 		// one `Branch protection rules`. All settings may be disabled,
 		// so it does not provide any guarantees.
-		protected := !(b.Protected != nil && !*b.Protected)
-		if !protected {
+		protected := (f.Values["branchProtected"] == 1)
+		if !protected && !contains(warnedBranches, branchName) {
 			dl.Warn(&checker.LogMessage{
-				Text: fmt.Sprintf("branch protection not enabled for branch '%s'", *b.Name),
+				Text: fmt.Sprintf("branch protection not enabled for branch '%s'", branchName),
 			})
+			warnedBranches = append(warnedBranches, branchName)
+		}
+		if branchName == "" {
+			e := sce.WithMessage(sce.ErrScorecardInternal, "probe is missing branch name")
+			return checker.CreateRuntimeErrorResult(name, e)
+		}
+
+		if _, ok := branchScores[branchName]; !ok {
+			branchScores[branchName] = &levelScore{}
+		}
+
+		var score, max int
+
+		switch f.Probe {
+		case blocksDeleteOnBranches.Probe, blocksForcePushOnBranches.Probe:
+			score, max = basicNonAdminProtection(f, dl)
+			branchScores[branchName].scores.basic += score
+			branchScores[branchName].maxes.basic += max
+
+		case dismissesStaleReviews.Probe, branchProtectionAppliesToAdmins.Probe:
+			score, max = adminThoroughReviewProtection(f, dl)
+			branchScores[branchName].scores.adminThoroughReview += score
+			branchScores[branchName].maxes.adminThoroughReview += max
+
+		case requiresApproversForPullRequests.Probe:
```

Comment:
This probe processes the `branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount` value. This value is used in two places in the evaluation. As such, we have a single probe that is used two places. The differentiating factor is the `"numberOfRequiredReviewers"` value.